tools/Dashboard/DevHome.Dashboard/Controls/WidgetControl.xaml.cs

```diff
@@ -18,21 +18,28 @@ public sealed partial class WidgetControl : UserControl
 {
     private MenuFlyoutItem _currentSelectedSize;
 
-    public WidgetControl()
-    {
-        this.InitializeComponent();
-        ActualThemeChanged += OnActualThemeChanged;
-    }
-
     public WidgetViewModel WidgetSource

```

Comment:
Does the `WidgetSource` change, or it's set by the parent once and remains unchanged? 
Why? Because `UpdateWidgetHeaderIconFill` is `async` and not awaited from the setter, hence we can potentially have concurrent execution of async methods.

Reply:
I believe it can change because GridView reuses its child elements. I think I was having trouble awaiting it, I'll look again...

Reply:
I'm not sure I'm finding a great way to deal with this. And TBH I'm not sure if it's something we really need to worry about. When I've seen these UI controls get reused, it's been when I remove and then add back a widget -- that's done by user actions and can't happen particularly fast. I'm not sure that this could ever happen concurrently in real life.